bindings/python/ckzg.c

```diff
@@ -10,19 +10,27 @@ static void free_KZGSettings(PyObject *c) {
 
 static PyObject* load_trusted_setup_wrap(PyObject *self, PyObject *args) {
   PyObject *f;
+  FILE *fp;
 
   if (!PyArg_ParseTuple(args, "U", &f))
     return PyErr_Format(PyExc_ValueError, "expected a string");
 
   KZGSettings *s = (KZGSettings*)malloc(sizeof(KZGSettings));
-
   if (s == NULL) return PyErr_NoMemory();
 
-  if (load_trusted_setup_file(s, fopen(PyUnicode_AsUTF8(f), "r")) != C_KZG_OK) {
+  fp = fopen(PyUnicode_AsUTF8(f), "r");
+  if (fp == NULL) {
+    free(s);
+    return PyErr_Format(PyExc_RuntimeError, "error reading trusted setup");
+  }
+
+  if (load_trusted_setup_file(s, fp) != C_KZG_OK) {
     free(s);
+    fclose(fp);
     return PyErr_Format(PyExc_RuntimeError, "error loading trusted setup");
   }
 
+  fclose(fp);
```

Comment:
Seems like the convention in the other bindings is to do something like:
```
	fp := C.fopen(cTrustedSetupFile, cMode)
	if fp == nil {
		panic("error reading trusted setup")
	}
	ret := C.load_trusted_setup_file(&settings, fp)
	C.fclose(fp)
	if ret == C.C_KZG_OK {
		loaded = true
	}
```

where we defer the retval check of `load_trusted_setup_file()` so that we can unconditionally `fclose()` once. Should we also adapt this code to that convention?

I think we are also missing an fclose on the error path of the Rust bindings. I can make a new issue for this or handle it here.

Reply:
Yes, good suggestion. I will make this change. And yes to a new PR for the Rust bindings.

Reply:
Pushed at 40f637693

Reply:
Opened https://github.com/ethereum/c-kzg-4844/pull/402 for the Rust bindings.